src/libpriv/rpmostree-core.c

```diff
@@ -1760,22 +1783,60 @@ rpmostree_dnf_add_checksum_goal (GChecksum   *checksum,
       const char *action_str = convert_dnf_action_to_string (action);
       g_checksum_update (checksum, (guint8*)action_str, strlen (action_str));
 
+      /* For pkgs that were added from the pkgcache repo (e.g. local RPMs and replacement
+       * overrides), make sure to pick up the SHA256 from the pkg metadata, rather than what
+       * libsolv figured out (which is based on our chopped off fake RPM, which clearly will
+       * not match what's in the repo). This ensures that two goals are equivalent whether
+       * the same RPM comes from a yum repo or from the pkgcache. */
+      const char *reponame = NULL;
+      if (pkgcache)
+        reponame = dnf_package_get_reponame (pkg);
+      if (g_strcmp0 (reponame, HY_CMDLINE_REPO_NAME) == 0)
+        {
+          g_autofree char *cachebranch = rpmostree_get_cache_branch_pkg (pkg);
+          g_autofree char *cached_rev = NULL;
+          if (!ostree_repo_resolve_rev (pkgcache, cachebranch, TRUE, &cached_rev, error))
```

Comment:
Should be `FALSE` here for `allow_noent` right?

Reply:
Good catch!